packages/core/src/tests/VectorStoreIndex.test.ts

```diff
@@ -0,0 +1,130 @@
+import { VectorStoreIndex } from '../indices/vectorStore/VectorStoreIndex';
+import { TextNode, Document, BaseRetriever, BaseQueryEngine } from '../indices/vectorStore/VectorStoreIndex'; // corrected import statement
+import { mockEmbeddingModel } from '../mocks/mockOpenAI'; // import the mockEmbeddingModel
+
+describe('VectorStoreIndex', () => {
+  test('init', async () => {
+    // Prepare inputs
+    const options = {
+      storageContext: new StorageContext(),
+      serviceContext: new ServiceContext(),
+      docStore: new BaseDocumentStore(),
+      vectorStore: new VectorStore(),
+      indexStruct: new IndexStruct(),
+    }; // fill with appropriate data
+
+    // Call the method
+    const result = await VectorStoreIndex.init(options);
+
+    // Assert the result
+    expect(result).toBeInstanceOf(VectorStoreIndex);
+    // Add more assertions as necessary
+
+    // Handle edge cases and errors
+    // Add code to handle edge cases and errors
+  });
+
+  // Write similar test blocks for the other methods: getNodeEmbeddingResults, buildIndexFromNodes, fromDocuments, asRetriever, asQueryEngine
+
+  test('getNodeEmbeddingResults', async () => {
+    // Prepare inputs
+    const nodes = [
+      new TextNode({text: 'Apple'}),
+      new TextNode({text: 'Banana'}),
+      new TextNode({text: 'Cherry'}),
+      new TextNode({text: 'Date'}),
+      new TextNode({text: 'Elderberry'}),
+      new TextNode({text: 'Fig'}),
+      new TextNode({text: 'Grape'}),
+      new TextNode({text: 'Honeydew'}),
+      new TextNode({text: 'Iced Apple'}),
+      new TextNode({text: 'Jackfruit'})
+    ];
+    // Use the mockEmbeddingModel instead of the ServiceContext
+    const serviceContext = mockEmbeddingModel;
```

Comment:
This isn't correct. Check out CallbackManager.test.ts to see how the mockEmbeddingModel is used.